cwms-data-api/src/test/java/cwms/cda/api/project/ProjectLockDenyHandlerIT.java

```diff
@@ -0,0 +1,162 @@
+
+    // other lock tests at : https://bitbucket.hecdev.net/projects/CWMS/repos/hec-cwms-data-access/
+    // browse/hec-db-integration-testing/src/test/java/wcds/dbi/oracle/project/
+    // TestProjectLocks.java#475,492,537
+
+
+    @Test
+    void test_can_deny_revoke() throws SQLException {
+        CwmsDatabaseContainer<?> databaseLink = CwmsDataApiSetupCallback.getDatabaseLink();
+        final String[] lockId = {null};
+        final String[] lockId2 = {null};
+
+        String projId = "deny_Proj";
+        String appId = "deny_test";
+        String officeMask = OFFICE;
+        TestAccounts.KeyUser user1 = TestAccounts.KeyUser.SPK_NORMAL;
+        TestAccounts.KeyUser user2 = TestAccounts.KeyUser.SPK_NORMAL2;
+        int revokeTimeout = 10;
+
+        try {
+            // setup
+            databaseLink.connection(c -> {
+                DSLContext dsl = getDslContext(c, OFFICE);
+                ProjectDao prjDao = new ProjectDao(dsl);
+                ProjectLockDao lockDao = new ProjectLockDao(dsl);
+
+                Project testProject = buildTestProject(OFFICE, projId);
+                prjDao.create(testProject);
+
+                lockDao.allowLockRevokerRights(OFFICE, officeMask, projId, appId, user1.getName());
+                lockDao.allowLockRevokerRights(OFFICE, officeMask, projId, appId, user2.getName());
+
+                boolean dontRevoke = false;
+
+                ProjectLock req1 = new ProjectLock.Builder(OFFICE, projId, appId).build();
+                lockId[0] = lockDao.requestLock(req1, dontRevoke, revokeTimeout);
+                assertNotNull(lockId[0]);
+
+                assertFalse(lockId[0].isEmpty());
+                // now user1 has a lock.
+            });
+
+
+            // have user2 try to get the lock on another thread.
+            new Thread(() -> {
+                try {
+                    databaseLink.connection(c -> {
+                        DSLContext dsl = getDslContext(c, OFFICE);
+                        ProjectLockDao lockDao = new ProjectLockDao(dsl);
+
+                        long before = System.currentTimeMillis();
+
+                        // this should block until revokeTimeout if we don't do anything
+                        // but we are going to deny the revoke so it should return before 10s and
+                        // user2 should not get the lock
+                        ProjectLock req1 = new ProjectLock.Builder(OFFICE, projId, appId).build();
+                        lockId2[0] = lockDao.requestLock(req1, true, revokeTimeout);
+                        assertNull(lockId2[0]);
+
+                        long after = System.currentTimeMillis();
+                        long elapsed = after - before;
+                        assertFalse(Math.abs((1000 * revokeTimeout) - elapsed) < 1000);
+
+                    });
+                } catch (SQLException e) {
+                    throw new RuntimeException(e);
+                }
+            }).start();
```

Comment:
So it *probably* would never matter, but on the change this hangs, this thread should be set to Daemon before start.

That said if we're going to consider more threading in these tests we should probably create an executor service pool with a factory that does that.

Reply:
I set the thread to daemon.

Reply:
And added a timeout parameter